### ISSUE TITLE
Usage of int64 as return

### ISSUE THREAD

**Issue by a31amit** [diskfs/go-diskfs#129]
First, Thanks for this cool library! 

Question / Problem Statment : 

I was trying to use this library to get starting offset values of partitions inside disk image from `partition.GetStart()` 
```
func (p *Partition) GetStart() int64 {
	_, lss := p.sectorSizes()
	return int64(p.Start) * int64(lss)
}
```

I am not sure if I fully understand this or identifying a potential problem here due to the return value is `int64` which could have negative integer numbers. However when working with partitions, If I understand correctly, cannot have a negative starting point and the minimum starting point is `0` so should not this be the return of `uint64`? 

PS: I am a newbie in golang world. I could have missed something to understand.

**Comment by deitch** [comment]
You are correct, and if it were up to me, those would be `uint64`. However, most of the basic golang disk utilities work off of int64, e.g. see [File.ReadtAt()](https://pkg.go.dev/os#File.ReadAt) or [fs.FileInfo](https://pkg.go.dev/io/fs#FileInfo) or any of dozens of others.

I believe those came because the underlying C calls use `long` for the most part, e.g. [fseek](https://man7.org/linux/man-pages/man3/fseek.3.html). It just made sense to stick with what people have.

**Comment by a31amit** [comment]
Make sense, thanks for explaining.
